docs/apis/streaming/windows.md

```diff
@@ -459,42 +459,106 @@ ready for processing. This allows to get the benefit of incremental window compu
 the additional meta information that writing a `WindowFunction` provides.
 
 This is an example that shows how incremental aggregation functions can be combined with
-a `WindowFunction`.
+a `WindowFunction`.  The `FoldFunction`/`WindowFunction` example shows how to extract the
+ending event-time of a window of sensor readings that contain a timestamp, 
+and the `ReduceFunction`/`WindowFunctions` example shows how to do eager window
+aggregation (only a single element is kept in the window).
 
 <div class="codetabs" markdown="1">
 <div data-lang="java" markdown="1">
 {% highlight java %}
-DataStream<Tuple2<String, Long>> input = ...;
+DataStream<SensorReading> input = ...;
 
 // for folding incremental computation
 input
     .keyBy(<key selector>)
     .window(<window assigner>)
-    .apply(<initial value>, new MyFoldFunction(), new MyWindowFunction());
+    .apply(Long.MIN_VALUE, new MyFoldFunction(), new MyWindowFunction());
+
+/* ... */
+
+private static  class myFoldFunction implements FoldFunction<SensorReading, Long> {
+
+    public Long fold(Long acc, SensorReading s) {
+        return Math.max(acc, s.timestamp());
+    }
+}
+
+private static class MyWindowFunction implements WindowFunction<Long, Long, String, TimeWindow> {
+
+    public void apply(String key, TimeWindow window, Iterable<Long> timestamps, Collector<Long> out) {
+            out.collect(timestamps.iterator().next());
```

Comment:
Not sure if this is a good example. The same result could be achieved by a single `FoldFunction`. How about the `FoldFunction` counts the number of records and the `WindowFunction` emits a `Tuple3` of key, end time of window, and count?